Content.Server/GameTicking/Rules/RandomRuleSystem.cs

```diff
@@ -0,0 +1,79 @@
+using Content.Server.Administration.Logs;
+using Content.Server.Chat.Managers;
+using Content.Server.GameTicking.Rules.Components;
+using Content.Server.StationEvents;
+using Content.Shared.GameTicking.Components;
+using Robust.Shared.Prototypes;
+using Robust.Shared.Random;
+using Content.Shared.Storage;
+using Content.Server.StationEvents.Components;
+
+namespace Content.Server.GameTicking.Rules;
+
+public sealed class RandomRuleSystem : GameRuleSystem<RandomRuleComponent>
+{
+    [Dependency] private readonly IPrototypeManager _prototypeManager = default!;
+    [Dependency] private readonly IRobustRandom _random = default!;
+    [Dependency] private readonly IAdminLogManager _adminLogger = default!;
+    [Dependency] private readonly IChatManager _chatManager = default!;
+    [Dependency] private readonly IComponentFactory _compFact = default!;
+    [Dependency] private readonly EventManagerSystem _event = default!;
+
+    private string _ruleCompName = default!;
+    public override void Initialize()
+    {
+        base.Initialize();
+    }
+
+    protected override void Added(EntityUid uid, RandomRuleComponent component, GameRuleComponent gameRule, GameRuleAddedEvent args)
+    {
+        var selectedRules = EntitySpawnCollection.GetSpawns(component.SelectableGameRules, _random);
+
+        Random rnd = new Random();
+        int ruleQuant = rnd.Next(component.MinRules, component.MaxRules);
+
+        for (int i = 0; i < ruleQuant; i++)
+        {
+            var availableEvents = _event.AvailableEvents(); // handles the player counts and individual event restrictions, we need to do it each i incase it changes
+            string? slag = null;
+
+            foreach (var rule in selectedRules)
+            {
+                if (selectedRules == null)
+                    return;
```

Comment:
If this is nullable it should be returning above (but I'm pretty sure it's not) instead of inside the loop where it's most expensive.